src/extension.ts

```diff
@@ -34,7 +34,13 @@ let ignoreVMArgs = false;
 export function activate(context: ExtensionContext) {
   let storagePath = context.storagePath;
   if (!storagePath) {
-    storagePath = os.homedir() + "/.lsp4xml";
+    let preferencePath : string = workspace.getConfiguration().get('xml.storagePath');
+    if(preferencePath && preferencePath.length > 0) {
+      storagePath = preferencePath;
+    }
+    else {
+      storagePath = os.homedir() + "/.lsp4xml";
+    }
   }
   let logfile = path.resolve(storagePath + '/lsp4xml.log');
```

Comment:
the problem here is we need to separate the logs from the workspace (which belongs to context.storagePath), from the caching directory reusable by all lsp4xml instances (a maven schema, once downloaded for a particular directory, can be reused by another)